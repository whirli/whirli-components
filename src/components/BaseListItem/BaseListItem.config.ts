// @ts-ignore
import { PropType } from '@composition';
import * as PropHelpers from '@whirli-components/helpers/props';
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseListItem/BaseListItem.config';

// Package config
import { PropKeys, PropValues } from './BaseListItem.constants';
import type { PropTag } from './BaseListItem.types';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  [PropKeys.TAG]: {
    required: false,
    type: String as PropType<PropTag>,
    default: 'li',
  },
  ...config.props,
};
