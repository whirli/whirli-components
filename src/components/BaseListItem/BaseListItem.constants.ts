export const PropKeys: Record<string, string> = {
  TAG: 'tag',
};

export const PropValues: Record<string, string[]> = {
  [PropKeys.TAG]: ['li', 'dd', 'dt'],
};
