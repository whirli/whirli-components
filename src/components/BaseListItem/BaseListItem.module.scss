@import '../../styles/Breakpoints.module.scss';

// default styles
.list-item {
    display: flex;
}

// breakpoint styles
$packageBreakpointStyles: ();