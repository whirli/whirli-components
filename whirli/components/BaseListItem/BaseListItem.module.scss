@import '@whirli-components/components/BaseListItem/BaseListItem.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));