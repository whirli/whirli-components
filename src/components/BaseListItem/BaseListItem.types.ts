import { PropKeys, PropValues } from './BaseListItem.constants';

const propTag: string[] = PropValues[PropKeys.TAG];

export type PropTag = typeof propTag[number];
