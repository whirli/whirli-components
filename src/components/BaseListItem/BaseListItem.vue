<template>
  <component :is="props.tag" :class="wrapperClasses">
    <div :class="contentClasses"><slot /></div>
  </component>
</template>

<script setup lang="ts">
// Vue
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseListItem/BaseListItem.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseListItem.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const wrapperClasses = [styles['list-item'], makeClasses(ComponentStyles, props, styles)];
const contentClasses = [styles['list-item__content']];
</script>
