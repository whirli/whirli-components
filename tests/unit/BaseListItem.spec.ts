import { mount } from '@vue/test-utils';

import BaseListItem from '@/components/BaseListItem/BaseListItem.vue';

let wrapper: any;
const text = 'Hello world';

describe('BaseListItem.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = mount(BaseListItem, {
      props: {
        tag: 'li',
      },
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders the correct element', () => {
    expect(wrapper.find('li').exists()).toBeTruthy();
  });

  it('renders the correct text', () => {
    expect(wrapper.text()).toBe(text);
  });
});
