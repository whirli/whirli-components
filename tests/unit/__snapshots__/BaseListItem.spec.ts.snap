// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseListItem.vue - Single classes renders the component 1`] = `
<li
  class=""
>
  <div
    class=""
  >
    
    Hello world
    
  </div>
</li>
`;
